csm/core/services/usl.py

```diff
@@ -438,93 +298,56 @@ async def post_register_device(
             reason = 'UDS base URL is not valid'
             Log.error(reason)
             raise CsmInternalError(desc=reason)
-        # Let ``_initialize_lyve_pilot_s3_resources()`` propagate its exceptions
-        (lyve_pilot_s3_credentials, lyve_pilot_s3_account, lyve_pilot_iam_user,
-         lyve_pilot_bucket) = await self._initialize_lyve_pilot_s3_resources(
-            s3_account_service,
-            s3_iam_users_service,
-            s3_bucket_service,
-            s3_account_name,
-            s3_account_email,
-            s3_account_password,
-            iam_user_name,
-            iam_user_password,
-            bucket_name,
-        )
-        try:
-            register_device_params = self._format_lyve_pilot_register_device_params(
-                url, pin, self._token)
-            access_params = await self._format_lyve_pilot_access_params(
-                lyve_pilot_s3_account,
-                lyve_pilot_iam_user,
-            )
-            registration_body = {
-                'registerDeviceParams': register_device_params,
-                'accessParams': access_params,
-            }
-        except Exception as e:
-            await self._cleanup(
-                s3_account_service,
-                s3_iam_users_service,
-                s3_bucket_service,
-                lyve_pilot_s3_credentials,
-                lyve_pilot_s3_account.get('account_name'),
-                lyve_pilot_iam_user.get('user_name'),
-                lyve_pilot_bucket.get('bucket_name'),
-            )
-            reason = 'Error on registration body assembly'
-            Log.error(f'{reason}---{str(e)}')
-            raise CsmInternalError(desc=reason)
+        # TODO validate registration body
+        endpoint_url = str(uds_url) + '/uds/v1/registration/RegisterDevice'
+        # FIXME add relevant certificates to SSL context instead of disabling validation
+        async with ClientSession(connector=TCPConnector(verify_ssl=False)) as session:
+            Log.info(f'Start device registration at {uds_url}')
+            async with session.put(endpoint_url, json=registration_info) as response:
+                if response.status != 201:
+                    reason = 'Could not start device registration'
+                    Log.error(f'{reason}---unexpected status code {response.status}')
+                    raise CsmInternalError(desc=reason)
+            Log.info('Device registration in process---waiting for confirmation')
+            timeout_limit = time.time() + 60
+            while time.time() < timeout_limit:
+                try:
+                    async with session.get(endpoint_url) as response:
+                        if response.status == 200:
+                            Log.info('Device registration successful')
+                            break
+                        elif response.status == 203:
```

Comment:
This is different from pathfinder as pathfinder tries token first and then token+PIN. We always try with both token and PIN. Are we good here? Probably we don't need to be backward compatible.

Reply:
On technical grounds, I think we should check for this status code not only because it is described in the UDS spec but also because not testing it would result in a server-side error being returned to the client, which would be semantically incorrect since it is clear that the failure is in the client side which failed to provide a PIN.

In reality I understand that there is no way to provide the token from our UI as of today, but I think we should not take that into consideration when implementing the BE. For instance, if we ever decide to have the FE ask for the token and the PIN in separate, and the PIN is not required, this implementation will be enough. It is a low-cost change (<5 lines) that decreases coupling between FE and BE in a substantial manner.

Reply:
And, just to clarify, I didn't make this change in order to achieve backwards compatibility. As a matter of fact, status code 203 is not returned by UDS <1.2.0.